libc/string/string_tokpbrkstr.c

```diff
@@ -0,0 +1,645 @@
+	TEST_ASSERT_EQUAL_PTR(loremStr, strtok_r(loremStr, "ImDr", &rest));
+	TEST_ASSERT_EQUAL_PTR(&loremStr[3], strtok_r(NULL, "ImDr", &rest));
+	TEST_ASSERT_EQUAL_PTR(&loremStr[5], strtok_r(NULL, "ImDr", &rest));
+	TEST_ASSERT_EQUAL_PTR(&loremStr[7], strtok_r(NULL, "ImDr", &rest));
+	TEST_ASSERT_EQUAL_PTR(&loremStr[11], strtok_r(NULL, "ImDr", &rest));
+	TEST_ASSERT_EQUAL_PTR(&loremStr[13], strtok_r(NULL, "ImDr", &rest));
+	TEST_ASSERT_EQUAL_PTR(NULL, strtok_r(NULL, "ImDr", &rest));
+
+
+	create_ascii_set(1);
+
+	/*
+	 * In this case, we tokenize the whole ASCII set using jump-over by
+	 * 2 elements. We are forced to jump over 2 places because strtok set
+	 *  divider placed as a null term and in this case, we are unable to
+	 * cover the whole set. Jumping over 2 places guarantees us output in the form
+	 * of one element form ascii set to do an assert with the place where
+	 * strtok has saved the stopping place
+	 */
+	TEST_ASSERT_EQUAL_PTR(asciiStr, strtok_r(asciiStr, "\2", &rest));
+	for (i = 4; i < sizeof(asciiStr) - 1; i += 2) {
+		separator[0] = i;
+		TEST_ASSERT_EQUAL_PTR(&asciiStr[i - 2], strtok_r(NULL, &separator[0], &rest));
+	}
+#endif
+}
+
+
+TEST(string_tok_r, out_of_ascii)
+{
+#ifdef __phoenix__
+	TEST_IGNORE();
+#else
+
+	create_extAscii_set();
+
+	char divider[BUFFSIZE],
+		separator[ASCII_LENGTH],
+		*rest = NULL;
+	int i;
+
+	sprintf(divider, "%c%c%c%c%c", extAsciiStr[0], extAsciiStr[64], extAsciiStr[128], extAsciiStr[192], extAsciiStr[255]);
+
+	/* Checking ability for tokenize element above standard charset */
+	TEST_ASSERT_EQUAL_PTR(&extAsciiStr[1], strtok_r(extAsciiStr, divider, &rest));
+	TEST_ASSERT_EQUAL_PTR(&extAsciiStr[65], strtok_r(NULL, divider, &rest));
+	TEST_ASSERT_EQUAL_PTR(&extAsciiStr[129], strtok_r(NULL, divider, &rest));
+	TEST_ASSERT_EQUAL_PTR(&extAsciiStr[193], strtok_r(NULL, divider, &rest));
+	TEST_ASSERT_EQUAL_PTR(NULL, strtok_r(NULL, divider, &rest));
+
+	create_extAscii_set();
+	/*
+	 * Checking available tokenization on all elements
+	 * from outside of ASCII charset one by one, with
+	 * the first separator set as the first element of the set
+	 */
+	TEST_ASSERT_EQUAL_PTR(extAsciiStr, strtok_r(extAsciiStr, "\128", &rest));
+	for (i = ASCII_LENGTH + 1; i < sizeof(separator); i++) {
+		separator[i - ASCII_LENGTH] = i;
+		TEST_ASSERT_EQUAL_PTR(&extAsciiStr[i - ASCII_LENGTH], strtok_r(NULL, &separator[i - ASCII_LENGTH], &rest));
+	}
+
+#endif
+}
+
+
+TEST(string_tok_r, restriction)
+{
+#ifdef __phoenix__
+	TEST_IGNORE();
+#else
+
+	char *rest[ASCII_LENGTH] = { 0 };
+	int i;
+
+	create_ascii_set(1);
+
+	for (i = 1; i < ASCII_LENGTH; i++) {
+
+		TEST_ASSERT_NOT_NULL(strtok_r(asciiStr, &asciiStr[i], &rest[i]));
+		TEST_ASSERT_NOT_NULL(rest[i]);
+	}
+#endif
+}
+
+
+/*
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////
+*/
+
+
+TEST_SETUP(string_str)
```

Comment:
Please look at the `strstr` and `strpbrk` tests after applying the suggestions mentioned before and make sure that there is nothing to improve in a similar way.